Framework/Algorithms/src/CalculateFlatBackground.cpp

```diff
@@ -253,9 +253,9 @@ void CalculateFlatBackground::convertToDistribution(
                            ? 1
                            : workspace->getNumberHistograms();
 
-  MantidVec adjacents(workspace->readX(0).size() - 1);
+  MantidVec adjacents(workspace->x(0).size() - 1);
   for (std::size_t i = 0; i < total; ++i) {
-    MantidVec X = workspace->readX(i);
+    auto &X = workspace->x(i);
     // Calculate bin widths
     std::adjacent_difference(X.begin() + 1, X.end(), adjacents.begin());
```

Comment:
The following is another example of something we might provide in `HistogramData`, i.e., we should gather this in some sort of notes, so we do not forget.